src/lib/mongoosePlugin.ts

```diff
@@ -7,7 +7,16 @@ import { IChangeEvent, IParams } from '../types/plugin';
 let client: MongoClient;
 const init = async (connectionString: string) => {
     if (!connectionString) { throw new Error('Connection string is required'); }
-    client = await new MongoClient(connectionString, { useNewUrlParser: true });
+    client = await new MongoClient(connectionString,
+        {
+            reconnectTries: 2,
+            useNewUrlParser: true,
+        });
+    client.connect((error: MongoError, mongoClient: MongoClient) => {
```

Comment:
Should be converted to a promise. This function will return before the connection is completed